### ISSUE TITLE
MOVE and alignChildren:false do not work as expected for non-card widgets

### ISSUE THREAD

**Issue by 96LawDawg** [ArnoldSmith86/virtualtabletop#993]
Look at the new MOVE tutorial.  The widgets in the far-right area are moved one at a time to a holder with alignChildren: true.  That makes them stack.  If you set alignChildren:false, I would have expected the widgets to show up in the new holder in relatively the same position they do in the current holder.  Instead, they get moved to the room absolute coordinates that were the equivalent of the old holder relative coordinates.

**Comment by 96LawDawg** [comment]
Forgot to mention that although this isn't related to #373, it would seem efficient to resolve those together.

**Comment by robartsd** [comment]
#373 is not related and is not likely to touch any relevant code.

MOVE simply calls `moveToHolder()` on the widget. `moveToHolder()` calls `this.checkParent(true)` which sets `parent` to `null`. When execution returns to `moveToHolder()` the widget is functionally at the room coordinates. `moveToHolder()` then sets the new `parent`. Eventually code execution reaches `holder.onChildAddAlign()` which translates the room coordinates to the relative coordinates within the holder. I don't know why this would not behave the same for cards as other widgets.

**Comment by 96LawDawg** [comment]
It does behave the same for cards.  I just hadn't tested that.  Cards are usually in piles and/or aligned in holders in my experience.

**Comment by ArnoldSmith86** [comment]
I honestly have no idea what I expect. The `MOVE` is asked not to align the widget but it has to be put _somewhere_.

**Comment by robartsd** [comment]
> I honestly have no idea what I expect.

I expect `MOVE` to behave exactly as `SET` `parent` unless the new parent widget has `alignChildren:true`.

**Comment by 96LawDawg** [comment]
> I expect `MOVE` to behave exactly as `SET` `parent` unless the new parent widget has `alignChildren:true`.

Yes.  That is it exactly.  More artfully said than I did.  But of course that is not happening because set parent would move it to the new holder relative position and not the room absolute position.

**Comment by 96LawDawg** [comment]
Drawing on something Ingegneus said (https://discord.com/channels/770758631146782780/795657499520663622/918051298950119436), I tested MOVE from one basic widget to another basic widget.  It works just like SET parent would.

**Comment by robartsd** [comment]
> Drawing on something Ingegneus said (https://discord.com/channels/770758631146782780/795657499520663622/918051298950119436), I tested MOVE from one basic widget to another basic widget. It works just like SET parent would.

I'm not sure if the key is moving from a basic widget or moving to a basic widget. I'm pretty sure a holder on one side of that would work. I think the key is figuring out where `onChildAddAlign()` is being called from.

Looking at the code, I see that #903 failed to correct for position of a card being dropped into a holder that is both transformed and has a stack offset set. `onChildAddAlign()` needs similar changes in holder.js as were added to widget.js.

**Comment by bjalder26** [comment]
I think this is a similar or the same issue as #676.

I opened a PR to try to address that (#740), but was very unsure whether or not I was going down the right path.

At the time, I thought when MOVE moves a widget to a holder with alignChildren set to false, that the parent should change, but the widget should remain in the same place in the room.  However the argument that the widget should have its relative position shifted to that of the holder is also a compelling argument.

Long story short my PR stalled because I wasn't sure what would be the 'right way' to do it, and now I am probably even less sure.  Maybe it should be scrapped for a totally different approach, but it might be helpful to see the issues I ran into.
